packages/components/src/Rating/Rating.tsx

```diff
@@ -0,0 +1,70 @@
+import { type ReactNode, type SyntheticEvent } from 'react';
+import {
+  type IconContainerProps as MuiIconContainerProps,
+  type RatingProps as MuiRatingProps,
+} from '@mui/material';
+
+import { DEFAULT_HINTS } from './constants';
+import { IconContainer } from './IconContainer';
+import { StyledRating } from './styles';
+
+export type RatingProps = Omit<MuiRatingProps, 'size' | 'onChange'> & {
+  /**
+   * Кастомные иконки
+   */
+  icons?: ReactNode[];
+
+  /**
+   * Кастомные подсказки
+   */
+  hints?: string[];
+
+  /**
+   * Флаг, отвечающий за отображение подсказок
+   * @default false
+   */
+  isVisibleHints?: boolean;
+
+  /**
+   * Функция, вызывающаяся при изменении значения
+   * @param value - выбранное значение
+   */
+  onChange: (value: number | null) => void;
+};
+
+// Компонент не экспортируется из пакета и пока не предназначен для использования в продуктах
+// Это связано с тем что он пока не стандартизирован на стороне дизайна и сейчас необходим для реализации компонента обратной связи
+export const Rating = ({
+  isVisibleHints,
+  icons,
+  hints = DEFAULT_HINTS,
+  onChange,
+  ...props
+}: RatingProps) => {
+  const handleChange = (
+    _: SyntheticEvent<Element, Event>,
+    value: number | null,
+  ) => onChange(value);
+
+  const renderIconContainer = ({
+    value,
+    ...iconContainerProps
+  }: MuiIconContainerProps) => (
+    <IconContainer
+      {...iconContainerProps}
+      value={value}
+      isVisibleHints={isVisibleHints}
+      icon={icons ? icons[value - 1] : null}
+      hint={hints[value - 1]}
+    />
+  );
+
+  return (
+    <StyledRating
+      IconContainerComponent={renderIconContainer}
+      getLabelText={(value: number) => hints[value - 1]}
```

Comment:
https://kaluga-astral.github.io/style-guide/docs/rules/react/logic#%D0%B7%D0%B0%D0%BF%D1%80%D0%B5%D1%89%D0%B5%D0%BD%D0%B0-%D1%80%D0%B5%D0%B0%D0%BB%D0%B8%D0%B7%D0%B0%D1%86%D0%B8%D1%8F-%D0%BE%D0%B1%D1%80%D0%B0%D0%B1%D0%BE%D1%82%D1%87%D0%B8%D0%BA%D0%BE%D0%B2-%D0%B2-jsx